### DISCUSSION TITLE
How is svelte being server side rendered?

### DISCUSSION THREAD

**Discussion by Bishwas-py** [livebud/bud#248] [Q&A]
This project is extremely interesting and awesome. But I wanna know how the svelte code is being server-side rendered here.

**Comment by Bishwas-py** [comment]
@matthewmueller Sorry to mention you, but I really wanna know this...

**Reply by matthewmueller** [reply]
Hey @Bishwas-py! We take the following steps:

1. Pre-bundle the Svelte compiler using [ESBuild](https://esbuild.github.io/api/)
2. Pre-load the bundled Svelte compiler into Go using [GoV8](https://github.com/rogchap/v8go)

Then for the views:

1. Use the preloaded Svelte compiler to turn Svelte code into JS
2. Run that JS through GoV8 to get HTML, CSS, JS
3. Serve the HTML to the browser

Does this help?

**Reply by Bishwas-py** [reply]
> * Pre-bundle the Svelte compiler using [ESBuild](https://esbuild.github.io/api/)
> * Pre-load the bundled Svelte compiler into Go using [GoV8](https://github.com/rogchap/v8go)

What is this for? As you have said the below one is for views, why have you mentioned these two points?

**Reply by Isotop7** [reply]
I dont know for sure but would except the svelte code has to be compiled to static html (+ js) on runtime like inserting model values etc. This static html is then served to the client.

**Reply by Bishwas-py** [reply]
> I dont know for sure but would except the svelte code has to be compiled to static html (+ js) on runtime like inserting model values etc. This static HTML is then served to the client. @matthewmueller @Isotop7

How's the context sent to the rendered HTML? Lemme know if I am understanding well...

First: Rendering all Svelte code to Javascript/CSS with `svelte/compiler`
Second: Converting the rendered JS into HTML
Third: Passing Go context into the rendered HTML?

How the context is passed to HTML, the looping and data spreading?
